src/aktualizr_secondary/aktualizr_secondary.cc

```diff
@@ -200,3 +176,60 @@ void AktualizrSecondary::connectToPrimary() {
     LOG_INFO << "Failed to connect to Primary";
   }
 }
+
+bool AktualizrSecondary::doFullVerification(const Metadata& metadata) {
+  // 5.4.4.2. Full verification  https://uptane.github.io/uptane-standard/uptane-standard.html#metadata_verification
+
+  // 1. Load and verify the current time or the most recent securely attested time.
+  //
+  //    We trust the time that the given system/OS/ECU provides, In ECU We Trust :)
+  TimeStamp now(TimeStamp::Now());
+
+  // 2. Download and check the Root metadata file from the Director repository, following the procedure in
+  // Section 5.4.4.3. DirectorRepository::updateMeta() method implements this verification step, certain steps are
+  // missing though. see the method source code for details
+
+  // 3. NOT SUPPORTED: Download and check the Timestamp metadata file from the Director repository, following the
+  // procedure in Section 5.4.4.4.
+  // 4. NOT SUPPORTED: Download and check the Snapshot metadata file from the Director repository, following the
+  // procedure in Section 5.4.4.5.
+  //
+  // 5. Download and check the Targets metadata file from the Director repository, following the procedure in
+  // Section 5.4.4.6. DirectorRepository::updateMeta() method implements this verification step
+  //
+  // The followin steps of the Director's target metadata verification are missing in DirectorRepository::updateMeta()
+  //  6. If checking Targets metadata from the Director repository, verify that there are no delegations.
+  //  7. If checking Targets metadata from the Director repository, check that no ECU identifier is represented more
```

Comment:
Can you make a tech debt ticket for this and put it in the robustness epic?

It looks like you've already implemented the check for the secondary; is your proposal to move that check into the Uptane code directly so it is applied universally as early as is reasonable?

Reply:
Sure, I'll do it.

I did #7, not #6, and yes my proposal to do apply it universally.

Reply:
Understood. We implicitly do step 6 in that we ignore the concept of delegations from the Director, but it's not a bad idea to check for them and throw a warning if found. Since we don't use the Snapshot role on the Director, though, all we can do is just check for them in the Targets metadata.

Reply:
https://saeljira.it.here.com/browse/OTA-4118